gvm/protocols/gmpv9/gmpv9.py

```diff
@@ -1093,6 +1117,29 @@ def __create_config(
         cmd.add_element("usage_type", usage_type.value)
         return self._send_xml_command(cmd)
 
+    def __create_config_from_osp_scanner(
+        self,
+        scanner_id: str,
+        name: str,
+        usage_type: UsageType,
+        function: str,
+        *,
+        comment: Optional[str] = None
+    ) -> Any:
+        if not name:
+            raise RequiredArgument(function=function, argument='name')
+
+        if not scanner_id:
+            raise RequiredArgument(function=function, argument='scanner_id')
+
+        cmd = XmlCommand("create_config")
+        if comment is not None:
+            cmd.add_element("comment", comment)
+        cmd.add_element("scanner", scanner_id)
+        cmd.add_element("name", name)
+        cmd.add_element("usage_type", usage_type.value)
+        return self._send_xml_command(cmd)
```

Comment:
Isn't it possible to add this into the existing `__create_config()` function with some conditions?
```
if usage_type == UsageType.SCAN:
  cmd.add_element("scanner", scanner_id)
else:
  cmd.add_element("copy", config_id)
```

?

Reply:
UsageType is if this config is scan or policy config not if it is generated from a existing config or osp scanner.

Using the __create_config() was my idea too but there are some problems with that.
We would need to include some parameter to toggle "scanner" instead of "copy". Another problem is that the scanner_id is required, but for getting the config from an osp scanner we need a scanner_id. In this case the scanner_id would be required. We could just use the config_id parameter and supply a osp scanner id instead. But then the meaning is not clear and we added implicit logic (config_id is not really a config_id but a scanner_id and so on).

Reply:
Ahh yes. I see. So we do it this way ...